slsim/lsst_science_pipeline.py

```diff
@@ -745,6 +745,106 @@ def dp0_time_series_images_data(butler, center_coord, radius="0.1", band="i", si
     return table_data
 
 
+def opsim_time_series_images_data(
+    butler, center_coord, radius="0.1", band="i", size=101
+):
+    """Creates time series data from opsim database.
+
+    :param butler: butler object
+    :param center_coord: A coordinate point around which we need to create time series
+        images.
+    :param radius: radius for query
+    :param band: imaging band
+    :param size: cutout size of images
+    :return: An astropy table containg time series images and other information
+    """
+
+    expo_information = tap_query(center_coords=center_coord, radius=radius, band=band)
+    calexp_image = list_of_calexp(expo_information, butler=butler)
+    radec = dp0_center_radec(calexp_image[0])
+    # Question: is radec_list only for 1 observation? Or can it be a list with multiple ra and decs?
```

Comment:
Here, radec is only for 1 observation. During the multiple lens injection, one has to loop this function through list of radec for multiple observations and create a list of astropy table for all observations. Then, one need to pass this list to the multiple_variable_lens_injection() function. So, if opsim is faster with list of radec, I think you can write a function with radec list and pass its output to multiple_variable_lens_injection() directly.

For single radec, you can write another function or give an option within the same function.